src/ListTabulator/index.ts

```diff
@@ -366,353 +445,450 @@ export default class ListTabulator {
+      return;
+    }
+
+    const currentItemContent = currentItem.querySelector<HTMLElement>(`.${DefaultListCssClasses.itemContent}`);
+
+    let endingHTML: string;
 
     /**
-     * Reverse the array to insert items
+     * If current item has no content, we should pass an empty string to the next created list item
      */
-    currentItemSublistItems.reverse().forEach((item) => {
+    if (currentItemContent === null) {
+      endingHTML = '';
+    } else {
       /**
-       * Check if we need to save the indent for current item children
-       *
-       * If this is the first item in the list then place its children to the same level as currentItem.
-       * Same as shift+tab for all of these children.
-       *
-       * If there is a previous sibling then place children right after target item
+       * On other Enters, get content from caret till the end of the block
+       * And move it to the new item
        */
-      if (!previousItem) {
-        /**
-         * The first item in the list
-         *
-         * Before:
-         * 1. Hello
-         *   1.1. |My
-         *     1.1.1. Wonderful
-         *     1.1.2. World
-         *
-         * After:
-         * 1. Hello|My
-         *   1.1. Wonderful
-         *   1.2. World
-         */
-        currentItem.after(item);
-      } else {
-        /**
-         * Not the first item
-         *
-         * Before:
-         * 1. Hello
-         *   1.1. My
-         *   1.2. |Dear
-         *     1.2.1. Wonderful
-         *     1.2.2. World
-         *
-         * After:
-         * 1. Hello
-         *   1.1. My|Dear
-         *   1.2. Wonderful
-         *   1.3. World
-         */
-        targetItem.after(item);
-      }
-    });
+      endingHTML = getContenteditableSlice(currentItemContent, currentNode, offset, 'right', true);
+    }
+
+    const itemChildren = currentItem?.querySelector(
+      `.${DefaultListCssClasses.itemChildren}`
+    );
 
     /**
-     * Remove current item element
+     * Create the new list item
      */
-    currentItem.remove();
+    const itemEl = this.renderer!.renderItem(endingHTML, { checked: false });
 
     /**
-     * Restore the caret position
+     * Move new item after current
      */
-    this.caret.restore();
-  }
+    currentItem?.after(itemEl);
 
+    /**
+     * If current item has children, move them to the new item
+     */
+    if (itemChildren) {
+      itemEl.appendChild(itemChildren);
+    }
+
+    this.focusItem(itemEl);
+  }
 
   /**
-   * Reduce indentation for current item
-   *
-   * @param {KeyboardEvent} event - keydown
-   * @returns {void}
+   * Method that is used for merging current item with previous one
+   * Content of the current item would be appended to the previous item
+   * Current item children would not change nesting level
+   * @param currentItem - current item html element
    */
-  shiftTab(event: KeyboardEvent): void {
+  mergeCurrentItemWithPrevious(currentItem: HTMLElement): void {
```

Comment:
why current? it accepts any item and merges it.

Also, I'm suggesting to create a type alias for a list item. The code will become more readable

```ts
type ListItemElement = HTMLElement;
```

```suggestion
  mergeItemWithPrevious(item: ListItemElement): void {
```